app/src/lib/stores/app-store.ts

```diff
@@ -1587,17 +1587,36 @@ export class AppStore {
     }
   }
 
+  private getLocalBranch(
+    repository: Repository,
+    branch: string
+  ): Branch | null {
+    const gitStore = this.getGitStore(repository)
+    return (
+      gitStore.allBranches.find(b => b.nameWithoutRemote === branch) || null
+    )
+  }
+
   /** This shouldn't be called directly. See `Dispatcher`. */
   public async _checkoutBranch(
     repository: Repository,
-    name: string
+    branch: Branch | string
   ): Promise<Repository> {
     const gitStore = this.getGitStore(repository)
     const kind = 'checkout'
 
+    const foundBranch =
+      branch instanceof String
```

Comment:
We should be using `typeof(branch) === "string"` since strings are a JavaScript primitive. I believe it's recommended to use `instanceof` for complex or custom types.